autoload/direnv/edit.vim

```diff
@@ -0,0 +1,42 @@
+" direnv.vim - support for direnv <http://direnv.net>
+" Author:       JINNOUCHI Yasushi <me@delphinus.dev>
+" Version:      0.2
+
+let s:direnv_edit_mode = get(g:, 'direnv_edit_mode', 'edit')
+
+function! direnv#edit#envrc() abort
+  if $DIRENV_DIR !=# ''
+    let l:envrc_dir = substitute($DIRENV_DIR, '^-', '', '')
+  elseif expand('%') ==# ''
+    let l:envrc_dir = getcwd()
+  else
+    let l:envrc_dir = expand('%:p')
+  endif
+  let l:envrc = l:envrc_dir . '/.envrc'
+  if !filereadable(l:envrc)
+    echom 'new .envrc file will be created:' l:envrc_dir
+  endif
+  call direnv#edit#execute(l:envrc)
+endfunction
+
+function! direnv#edit#direnvrc() abort
+  if $XDG_CONFIG_HOME ==# ''
+    let l:direnvrc_dir = $HOME . '/.config/direnv'
+  else
+    let l:direnvrc_dir = $XDG_CONFIG_HOME . '/direnv'
+  endif
+  if filereadable(l:direnvrc_dir . '/direnvrc') ||
+        \ isdirectory(l:direnvrc_dir) && !filereadable($HOME . '/.direnvrc')
+    let l:direnvrc = l:direnvrc_dir . '/direnvrc'
+  else
+    let l:direnvrc = $HOME . '/.direnvrc'
+  endif
+  if !filereadable(l:direnvrc)
+    echom 'new direnvrc file will be created:' l:direnvrc
```

Comment:
this will default to `~/.direnvrc`, which is the old and deprecated path

Reply:
Ah, is it deprecated? I will fix the whole logic. It should not create `~/.direnvrc` implicitly.

For users who have no `~/.config/direnv` directory, Vim should ask to make it before opening the buffer, I think. Without that, Vim causes an error in writing the content. 🤔

Reply:
1. When `~/.config/direnv/direnvrc` is readable, it opens that.
2. When `~/.direnvrc` is readable, it opens that.
3. When `~/.config/direnv` exists, it opens `~/.config/direnv/direnvrc`.
4. Otherwise, it will ask the user to create `~/.config/direnv` directory. If OK, it creates the dir, and open `~/.config/direnv/direnvrc`.

It may not have to ask and can create the dir always. What do you think? @zimbatm

Reply:
I think it's fine if the directory creation is implicit for point (4). No need to ask the user about it.